drivers/interrupt_controller/intc_ite_it8xxx2.c

```diff
@@ -133,6 +134,11 @@ void ite_intc_irq_disable(unsigned int irq)
 	/* critical section due to run a bit-wise OR operation */
 	unsigned int key = irq_lock();
 	CLEAR_MASK(*en, BIT(i));
+	/*
+	 * This load operation will guarantee the above modification of
+	 * SOC's register can be seen by any following instructions.
+	 */
+	_ier = *en;
```

Comment:
Is a `fence` instruction more appropriate to use here?

Reply:
Barrier instruction can not synchronize chip register, so we introduce workaround here.